tests/transparency.rs

```diff
@@ -76,6 +76,42 @@ fn inbound_tcp() {
     assert_eq!(tcp_client.read(), msg2.as_bytes());
 }
 
+#[test]
+fn outbound_kafka() {
```

Comment:
The tests in the `tests` directory are intended as end-to-end tests of the whole proxy. I think this test is focused enough that it might eventually be better off as a test specifically of the `Protocol::detect` method. In that case, it would live in that module, and the test could be much simpler, since you wouldn't have to create an entire proxy, a tcp client, and so on --- you could just call `Protocol::detect(KAFKA_REQUEST_API_VERSIONS)` and assert that you get `Some(Protocol::Kafka)` back.

With that said, having an end-to-end test for kafka proxying and detection is probably also valuable...